logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/MLSTransportProvider.kt

```diff
@@ -0,0 +1,97 @@
+import com.wire.kalium.persistence.dao.message.LocalId
+
+interface MLSTransportProvider : MLSTransporter
+
+internal class MLSTransportProviderImpl(
+    private val selfUserId: UserId,
+    private val mlsMessageApi: MLSMessageApi,
+    private val localEventRepository: LocalEventRepository,
+    private val mlsCommitBundleMapper: MLSCommitBundleMapper = MapperProvider.mlsCommitBundleMapper(),
+) : MLSTransportProvider {
+
+    override suspend fun sendMessage(mlsMessage: ByteArray): MlsTransportResponse {
+//         return messageRepository.sendMLSMessage(MLSMessageApi.Message(mlsMessage))
+//             .onSuccess { response ->
+//                 // handleMlsRecipientsDeliveryFailure(response, response) TODO we don't have an access to conversationId and messageId
+//             }
+//             .fold({
+//                 if (it is NetworkFailure.ServerMiscommunication && it.kaliumException is KaliumException.InvalidRequestError) {
+//                     MlsTransportResponse.Abort((it.kaliumException as KaliumException.InvalidRequestError).errorResponse.label)
+//                 } else {
+//                     MlsTransportResponse.Abort(it.toString())
+//                 }
+//             }, {
+//                 MlsTransportResponse.Success
+//             })
+        // TODO: currently not used by core crypto, leave error logger if it somehow gets used
+        kaliumLogger.e("[MLSTransportProvider] sendMessage triggered which should not happen")
+        return MlsTransportResponse.Success
+    }
+
+    override suspend fun sendCommitBundle(commitBundle: CommitBundle): MlsTransportResponse {
+        return wrapApiRequest {
+            mlsMessageApi.sendCommitBundle(mlsCommitBundleMapper.toDTO(commitBundle))
+        }
+            .onSuccess {
+                processCommitBundleEvents(it.events)
+            }.fold({
+                MlsTransportResponse.Abort(it.toString())
+            }, {
+                MlsTransportResponse.Success
+            })
+    }
+
+    private fun processCommitBundleEvents(events: List<EventContentDTO>) {
+        events.forEach { eventContentDTO ->
+            val event =
+                MapperProvider.eventMapper(selfUserId).fromEventContentDTO(
+                    LocalId.generate(),
+                    eventContentDTO
+                )
+            if (event is Event.Conversation) {
+                kaliumLogger.d("[MLSTransportProvider] Processing commit bundle events")
+                localEventRepository.emitLocalEvent(event)
+            }
+        }
+    }
+// TODO check if its needed
+//     private suspend fun handleMlsRecipientsDeliveryFailure(message: Message, messageSent: MessageSent) =
+//         if (messageSent.failedToConfirmClients.isEmpty()) Either.Right(Unit)
+//         else {
+//             messageRepository.persistRecipientsDeliveryFailure(message.conversationId, message.id, messageSent.failedToConfirmClients)
+//         }
```

Comment:
Only needed after we start sending application messages through the mls transporter.